test/framework/template.go

```diff
@@ -0,0 +1,89 @@
+package framework
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	releasev1 "github.com/aws/eks-anywhere/release/api/v1alpha1"
+)
+
+// ProviderTemplateNameGenerator is an interface for getting template name for each provider.
+type ProviderTemplateNameGenerator interface {
+	EnvVarForTemplate(osFamilyStr, eksDName string) string
+	DefaultNameForTemplate(osFamilyStr, eksDName string) string
+	SearchTemplate(ctx context.Context, template string) (string, error)
+	DefaultEnvVarForTemplate(osFamilyStr string, kubeVersion anywherev1.KubernetesVersion) string
+}
+
+// TemplateCache is a map of template name.
+type TemplateCache struct {
```

Comment:
I think this can be unexported as well
also, what do you think about

```suggestion
type templateRegistry struct {
```

This does more things that just caching values, it's also able to retrieve them
The cache is just an internal implementation detail of the registry